java/sage/EPG.java

```diff
@@ -651,161 +671,189 @@ public void run()
 
         long minWait = MAINTENANCE_FREQ - (Sage.time() - wiz.getLastMaintenance());
         if (minWait <= 0) minWait = 1;
+
         synchronized (sources)
         {
           for (int i = 0; (i < sources.size()) && alive; i++)
           {
+            if(Sage.getBoolean("wizard/scheduled_maintenance", false)){
+                //determine the wait until the next scheduled update time
+                //calc the next time as the user may have changed the schedule settings
+                nextScheuldedMaintenanceTime = getNextScheduledMaintenanceTime();
+                minWait = nextScheuldedMaintenanceTime - Sage.time();
```

Comment:
Isn't there a problem where if you set minWait in this block...then it can get overridden on line 687 with the one from the EPGDataSource?  So the scheduled maintenance only takes effect if the last EPGDataSource has an update time further out then the scheduled time.  It seems like this check should be outside of this loop to override the other update timing logic.